VerifyTests/Verify
# Add *.received.* to MSTest's results attachments

Fixes #1182

This change adds any *.received.* files from failing diffs to MSTest's results attachments. Attaching the received file makes it easy to download from a CI server's test results viewer without needing to modify the CI pipeline or YAML files to collect received files and upload them as build artifacts.

To accomplish that I:

1. Updated the `VerifyException` to track the `FilePair`s for mismatches
2. Wrap the inner verify to catch the exception and add the FilePair to the `TestContext`

In order to test the behavior, the logic is as follows:

1. Create a test where the *.received.* will not match the *.verified.* (Foo and Bar in this case)
2. Disable diffing so that unit test runs don't pop a diff tool
3. Create a custom `[TestMethod]` attribute, as that's the only way I know of to access the attachments
4. As part of `ReceivedFileRequiredTestMethodAttribute`'s execute, verify each test has a *.received.* file, and if not set the test result to failed
